### DISCUSSION TITLE
Autobreak long lines in markdown files doesn't work

### DISCUSSION THREAD

**Discussion by MatejBransky** [LazyVim/LazyVim#2512] [Q&A] [answered]
Hi, I've enabled LazyVim plugins formatting/prettier and lang/markdown. I don't know if it should work but I would like to automatically break long line in markdown files but it doesn't do anything with them. Just markdownlint is yelling at me that lines are too long so I have to manually break them. Is that expected in LazyVim or should I search for some config bug in my setup?

Example: it shows lint warning but the action format doesn't do anything with it.

<img width="1270" alt="image" src="https://github.com/LazyVim/LazyVim/assets/20944757/9aa6ad32-f44a-4266-a1fc-39a4cfb4cc74">

**Comment by dpetka2001** [comment]
You should set `vim.opt.textwidth = number` to whatever number you like in your `options.lua` file.

**Reply by MatejBransky** [reply]
Thank you for your response but nothing has changed. It still doesn't autobreak the line. 🤔

**Reply by MatejBransky** [reply]
Btw if I try to format via `<leader>cf` then it shows above the statusline "formatting null-ls". Is that expected? I thought that LazyVim uses `conform.nvim`..

**Reply by dpetka2001** [reply]
For me setting `textwidth` makes it break lines at the number I define (for Markdown files). What does `:set wrap?` show?

For your second question you must have the `none-ls` extra enabled and some kind of `none-ls` spec in your personal configuration. You should remove them if you don't want to use `none-ls` and use `conform` instead.

**Reply by MatejBransky** [reply]
> What does :set wrap? show?

Just "wrap" 
<img width="452" alt="image" src="https://github.com/LazyVim/LazyVim/assets/20944757/1d2530b0-82b4-45f9-bf35-1ff0c9e3b325">

> For your second question you must have the none-ls extra enabled and some kind of none-ls spec in your personal configuration. You should remove them if you don't want to use none-ls and use conform instead.

You're right. I had enabled `none-ls` in LazyExtras but when I disabled it looks like that nothing changed. Only "formatting null-ls" disappeared. Formatting itself works as before (autoindentation, reducing multiple blank lines, etc. but not the auto wrapping). And other than that I didn't found any "none-ls" via live grep in the nvim config folder.

**Reply by dpetka2001** [reply]
I just tried with your configuration with the latest changes you pushed in your dotfiles and it works perfectly well for me. Here's a screencast with your configuration

[Screencast 2024-02-06 13:01:37.webm](https://github.com/LazyVim/LazyVim/assets/12776461/e6ba630e-4088-42b5-9de1-65ab75dbaac9)

**Reply by MatejBransky** [reply]
🤦 Oh my god, I'm stupid. It happens only in list items. Common paragraphs works as expected. I wasn't trying a paragraph but only the list item starting with "-". What if you try that? Does it work too?

Example:
```md
### Tactical vs strategic programming

- Tactical approach - just coding and using whatever works best until it work the way we want to (PoC, experiments)
```

Weird. So the text is wrapped if I'm typing it but if I paste it then the "format" action doesn't break it to multiple lines. 🤔 I'm confused what's the issue then.

Btw I don't have any other file in the repository. It is just a Github gist. If I try to reproduce it in some common repository then the break-line action works as expected. 😕

**Reply by dpetka2001** [reply]
Yep totally fine

[Screencast 2024-02-06 13:23:26.webm](https://github.com/LazyVim/LazyVim/assets/12776461/12834c07-7499-43bc-be9a-a03f3c7a2449)

**Reply by MatejBransky** [reply]
Thank you for your time. I know that it works at least. But something weird is ignored in my Github gist and I don't know what it can be..but that's for another topic.

Take care, bye! 👋

**Reply by MatejBransky** [reply, marked as answer]
I've found it! It was caused by the ["prose wrap"](https://prettier.io/docs/en/options.html#prose-wrap) option which is "preserve" by default so long lines weren't broken to multiple lines. 🤦 To break them you have to use the value "always".

So I've added some default config to the projects folder and now all projects works as expected.

I'm glad that I know what was it. 😅

**Reply by dpetka2001** [reply]
@MatejBransky 

For the format the formatter you use is responsible. It has nothing to do with wrapping lines in `insert` mode.

In this case `prettier` is used for formatting Markdown files, so you would need a `.prettierrc` with following contents
```json
{
  "printWidth": 80,
  "proseWrap": "always"
}
```
Test this out. The option `proseWrap` was at fault here, since by default it is set to `preserve`. For more info see [here](https://prettier.io/docs/en/options.html#prose-wrap).

**Reply by dpetka2001** [reply]
Oh sorry, didn't see your previous comment cuz I had the tab already opened and didn't refresh correctly until after I answered as well. 

Glad you were able to resolve it :smile: .

**Reply by dpetka2001** [reply]
Please choose your own answer as suggested. You had already solved it on your own. It just doesn't sit right with me to accept mine :stuck_out_tongue:. If you'd like just write a new answer with a suggested `.prettierrc` like in my post. But in my opinion, it should be your answer the suggested one.

**Reply by redoxahmii** [reply]
Can this be disabled for headings too if they are long in length?
And one more issue is this 
 │     MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2] markdownlint [234, 1]

Whenever i save a file it creates a space underneath which then prompts this error. if i remove the space and save again it repeats the process.
